pkg/engine/imageVerify.go

```diff
@@ -153,15 +153,30 @@ func (iv *imageVerifier) verify(imageVerify *v1.ImageVerification, images map[st
 				continue
 			}
 
+			if imageVerify.DigestMutate == nil {
+				digest := true
+				imageVerify.DigestMutate = &digest
+			}
+
 			var ruleResp *response.RuleResponse
 			if len(imageVerify.Attestations) == 0 {
 				var digest string
 				ruleResp, digest = iv.verifySignature(imageVerify, imageInfo)
-				if ruleResp.Status == response.RuleStatusPass {
+				if imageInfo.Digest == "" && *imageVerify.DigestMutate && ruleResp.Status == response.RuleStatusPass {
 					iv.patchDigest(path, imageInfo, digest, ruleResp)
```

Comment:
should we handle errors from patch digest and fail the rule, if the digest cannot be mutated?

Reply:
yes! we should

Reply:
@JimBugwadia I tried to handle the errors and fail the rule let me know if that's the correct approach. Thanks!